routes/donations.js

```diff
@@ -30,24 +37,61 @@ router.get('/new', async (req, res) => {
 });
 
 // creates a new donation
-router.post('/', async (req, res) => {
+router.post('/new', multer1.upload, async (req, res) => {
   try {
+    const uploader = async (path) => await cloudinary.uploads(path, 'Images');
+    const urls = [];
+    const files = req.files;
+
+    for (const file of files) {
+      const { path } = file;
+      const newPath = await uploader(path);
+      urls.push(newPath);
+
+      //deleting file from server after upload
+      fs.unlinkSync(path);
+    }
+
+    if (files.length == 0) {
+      res.status(405).render('donations/new', {
+        title: 'error',
+
+        msg: 'Images not uploaded successfully, Please Select an Image',
+      });
+      return;
+    }
+    // res.status(200).render('donations/new', {
+    //   title: 'uploaded',
+
+    //   msg: 'Images uploaded successfully',
+    // });
+
+    let images = [];
+
+    for (let i = 0; i < urls.length; i++) {
+      images.push(urls[i].url);
+    }
+
+    console.log('inside post route');
+
     let { name, description, quantity, region, zipcode } = req.body;
     let newDonation = await donationData.create(
       name,
       description,
       quantity,
       region,
       zipcode,
-      req.session.user._id
+      images
+      // req.session.user._id
```

Comment:
@iamikenna Please add this line as well. We need to save donor id for donations.